File: Check Whether Two Strings are Almost Equivalent/Check_Whether_Two_Strings_are_Almost_Equivalent.py
class Solution:
    def checkAlmostEquivalent(self, word1: str, word2: str) -> bool:
        mpp = Counter(word1)
        mpp2 = Counter(word2)
        
        for c in set(mpp.keys()).union(mpp2.keys()):
            if abs(mpp.get(c, 0) - mpp2.get(c, 0)) > 3:
                return False
        
        return True
        

File: Check Whether Two Strings are Almost Equivalent/Check_Whether_Two_Strings_are_Almost_Equivalent.java
class Solution {
    public boolean checkAlmostEquivalent(String word1, String word2) {
        Map<Character, Integer> mpp = new HashMap<>(), mpp2 = new HashMap<>();
        for (char c : word1.toCharArray()) {
            mpp.put(c, mpp.getOrDefault(c, 0) + 1);
        }
        for (char c : word2.toCharArray()) {
            mpp2.put(c, mpp2.getOrDefault(c, 0) + 1);
        }
        for (Map.Entry<Character, Integer> entry : mpp.entrySet()) {
            if (Math.abs(entry.getValue() - mpp2.getOrDefault(entry.getKey(), 0)) > 3) {
                return false;
            }
            mpp.put(entry.getKey(),0);
            mpp2.put(entry.getKey(), 0);
        }
        for (Map.Entry<Character, Integer> entry : mpp2.entrySet()) {
            if (Math.abs(entry.getValue() - mpp.getOrDefault(entry.getKey(), 0)) > 3) {
                return false;
            }
        }
        return true;
    }
}